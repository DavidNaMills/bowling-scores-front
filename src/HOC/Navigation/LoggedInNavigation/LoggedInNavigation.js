import React from 'react';
import { NavLink } from 'react-router-dom';
import classes from '../Navigation.module.scss';
import ball from '../../../images/ball.png';

const LoggedInNavigation = (props) => (
    <div className={classes.navigation}>
        <div className={classes.navigation_section}>
            <NavLink exact className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/dashboard'><img src={ball} height='25' width='25' /></NavLink>
            <NavLink className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/game'>Play</NavLink>
        </div>
        <div className={classes.navigation_section}>
            <p className={classes.navigation__username} style={{ color: `rgb(${props.user.color})` }}>{props.user.username}</p>
            <NavLink className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/logout'>Logout</NavLink>
        </div>
    </div>
)

export default React.memo(LoggedInNavigation);