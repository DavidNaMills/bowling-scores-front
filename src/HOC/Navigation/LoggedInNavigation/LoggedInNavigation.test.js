import React from 'react';
import { NavLink, BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import LoggedInNavigation from './LoggedInNavigation';

describe('<LoggedIn /> test suite', () => {
    const user = { username: 'test test', color: '156, 156, 156' }
    const wrapper = mount(
        <BrowserRouter>
            <LoggedInNavigation user={user} />
        </BrowserRouter>
    );

    it('snapshot test', () => {
        const component = renderer.create(
            <BrowserRouter>
                <LoggedInNavigation user={user} />
            </BrowserRouter>
        );
        const app = component.toJSON();
        expect(app).toMatchSnapshot();
    });

    it('renders 3 <NavLink> components', () => {
        expect(wrapper.find(NavLink).length).toBe(3);
    });

    it('renders username', () => {
        expect(wrapper.find('p').length).toBe(1);
        expect(wrapper.find('p').text()).toEqual(user.username);
    });

    it('classNames are set', () => {
        expect(wrapper.find('.navigation_section').length).toBe(2);
    });

});