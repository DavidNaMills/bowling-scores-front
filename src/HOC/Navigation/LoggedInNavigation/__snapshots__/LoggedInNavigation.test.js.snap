// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoggedIn /> test suite snapshot test 1`] = `
<div
  className="navigation"
>
  <div
    className="navigation_section"
  >
    <a
      aria-current={null}
      className="navigation__link"
      href="/dashboard"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      aria-current={null}
      className="navigation__link"
      href="/game"
      onClick={[Function]}
    >
      Game
    </a>
  </div>
  <div
    className="navigation_section"
  >
    <p
      className="navigation__username"
      style={
        Object {
          "color": "rgb(156, 156, 156)",
        }
      }
    >
      test test
    </p>
    <a
      aria-current={null}
      className="navigation__link"
      href="/logout"
      onClick={[Function]}
    >
      Logout
    </a>
  </div>
</div>
`;
